interchaintest/fee_middleware_test.go

```diff
@@ -0,0 +1,199 @@
+package interchaintest_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	relayertest "github.com/cosmos/relayer/v2/interchaintest"
+	interchaintest "github.com/strangelove-ventures/interchaintest/v7"
+	"github.com/strangelove-ventures/interchaintest/v7/chain/cosmos"
+	ibc "github.com/strangelove-ventures/interchaintest/v7/ibc"
+	"github.com/strangelove-ventures/interchaintest/v7/testreporter"
+	"github.com/strangelove-ventures/interchaintest/v7/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestFeeMiddleware(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	t.Parallel()
+
+	// Get both chains
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{Name: "juno", ChainName: "chaina", Version: "v13.0.0", ChainConfig: ibc.ChainConfig{ChainID: "chaina", GasPrices: "0.0ujuno"}},
+		{Name: "juno", ChainName: "chainb", Version: "v13.0.0", ChainConfig: ibc.ChainConfig{ChainID: "chainb", GasPrices: "0.0ujuno"}}},
+	)
+
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+	chainA, chainB := chains[0].(*cosmos.CosmosChain), chains[1].(*cosmos.CosmosChain)
+
+	ctx := context.Background()
+	client, network := interchaintest.DockerSetup(t)
+
+	rf := relayertest.NewRelayerFactory(relayertest.RelayerConfig{InitialBlockHistory: 50})
+	r := rf.Build(t, client, network)
+
+	const pathChainAChainB = "chainA-chainB"
+
+	// Build the network
+	ic := interchaintest.NewInterchain().
+		AddChain(chainA).
+		AddChain(chainB).
+		AddRelayer(r, "relayer").
+		AddLink(interchaintest.InterchainLink{
+			Chain1:  chainA,
+			Chain2:  chainB,
+			Relayer: r,
+			Path:    pathChainAChainB,
+			CreateChannelOpts: ibc.CreateChannelOptions{
+				SourcePortName: "transfer",
+				DestPortName:   "transfer",
+				Order:          ibc.Unordered,
+				Version:        "{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}",
+			},
+			CreateClientOpts: ibc.DefaultClientOpts(),
+		})
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	require.NoError(t, ic.Build(ctx, eRep, interchaintest.InterchainBuildOptions{
+		TestName:         t.Name(),
+		Client:           client,
+		NetworkID:        network,
+		SkipPathCreation: false,
+	}))
+
+	t.Cleanup(func() {
+		_ = ic.Close()
+	})
+
+	err = testutil.WaitForBlocks(ctx, 10, chainA, chainB)
+	require.NoError(t, err)
+
+	// ChainID of ChainA
+	chainID_A := chainA.Config().ChainID
```

Comment:
nit: It's rare to see underscores in the names of variables in Go, could we remove that just for consistency?